src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -325,11 +327,22 @@ BlockIO InterpreterCreateQuery::createDatabase(ASTCreateQuery & create)
 
         if (!load_database_without_tables)
         {
-
             /// We use global context here, because storages lifetime is bigger than query context lifetime
             TablesLoader loader{getContext()->getGlobalContext(), {{database_name, database}}, mode};
-            loader.loadTables();
-            loader.startupTables();
+            auto load_tasks = loader.loadTablesAsync();
+            auto startup_tasks = loader.startupTablesAsync();
+            if (getContext()->getGlobalContext()->getServerSettings().async_load_databases)
```

Comment:
@serxa does it make sense to load databases asynchronously in case of ATTACH query? It may only surprise users I guess, and you can scale it by your own anyway (unlike server startup)

Reply:
Yes, now any load is asynchronous, we can either wait or not wait.

Reply:
In most cases, it is okay to do it async and not even wait, because all operations that require table/database to be loaded will wait for it.

Reply:
@serxa Do you have objections on making ATTACH queries non-async?

Reply:
No, I think we could change this to what it was before. It was just a way to do everything simple. But sync wait looks more like previous behaviour indeed.